src/common/vector_operations/vector_cast.cpp

```diff
@@ -198,6 +200,7 @@ static bool VectorStringCastNumericSwitch(Vector &source, Vector &result, idx_t
 		return VectorTryCastStrictLoop<string_t, uint64_t, duckdb::TryCast>(source, result, count, strict,
 		                                                                    error_message);
 	case LogicalTypeId::HUGEINT:
+	case LogicalTypeId::UUID:
```

Comment:
This case here should not be necessary, `UUID` is not a number and doesn't belong in this switch.